pkg/metrics/metrics.go

```diff
@@ -260,3 +266,17 @@ func (c *client) WriteMetricAsync(ctx context.Context, name string, count int64)
 func NoopWriter() MetricWriter {
 	return &client{OptOut: true}
 }
+
+// WithClient creates a new context with the provided client attached.
+func WithClient(ctx context.Context, client *client) context.Context {
+	return context.WithValue(ctx, metricsKey, client)
+}
+
+// FromContext returns the metrics client stored in the context.
+// If no such client exists, a default noop client is returned.
+func FromContext(ctx context.Context) MetricWriter {
+	if client, ok := ctx.Value(metricsKey).(MetricWriter); ok {
+		return client
+	}
+	return NoopWriter()
```

Comment:
I don't think you want to return a new instance of the NoopWriter. `DefaultLogger` is a singleton: https://github.com/abcxyz/pkg/blob/45258f93afbaca705ef0503c37ef88451ed45b02/logging/logger.go#L141-L143

Reply:
Changed to use Singleton pattern